### PR TITLE
Adding logger

### PR DESCRIPTION
**Pull request by babineaum**
need logger to print out comments when we have issues where we can't find "shipit" comment

### CODE REVIEW

File: github_pr.py [hunk 1]
```diff
@@ -441,9 +441,18 @@ def main():
     parser.add_argument('--condition-approved-mergers', default=None, nargs='+', help='list of usernames of approved mergers')
     parser.add_argument('--condition-approved-mergers-file', action='store_true', help='check a file for list of usernames of approved mergers')
     parser.add_argument('--approved-mergers-file-path', default='./MAINTAINERS.txt', help='location of file of usernames of approved mergers')
-
+    parser.add_argument("-v", "--verbose", const=1, default=0, type=int, nargs="?",
+                    help="increase verbosity: 0 = only warnings, 1 = info, 2 = debug. No number means info. Default is no verbosity.")
```
Comment (dx-pbuckley) [inline github_pr.py]: how about a more concise `help="logger verbosity: 0 = WARN (default), 1 = INFO, 2 = DEBUG"`

File: github_pr.py [hunk 2]
```diff
@@ -441,9 +441,18 @@ def main():
...
     args = vars(parser.parse_args())
 
+    logger = logging.getLogger()
```
Comment (dx-pbuckley) [inline github_pr.py]: do you need `import logging` at the top?

Comment (babineaum) [inline github_pr.py]: the way i saw it online was like this, but im not sure im doing it 100% correct. I thought i needed to instantiate a logging object (logger) to mod the levels for the numbers. Then i just use the logging the same as i would if it was just the logging lib. any input is appreciated. I did remove it and the code broke :)

File: github_pr.py (hunk 1 above)
Comment (babineaum) [inline github_pr.py]: switched this

File: github_pr.py (hunk 2 above)
Comment (dx-pbuckley) [inline github_pr.py]: You added it on line 9, that's what you needed

**Comment by ferrants** [comment]
nicee

**Comment by babineaum** [comment]
https://jenkins-staging.devaws.dataxu.net/job/tools_github-CD-checks/149/console

@cchristous moved that logger def and also ran a test with the verbose turned on. i am going to create a jjb ticket as well to add this verbose debugging to github-pr running on CD checks

File: github_pr.py [hunk 3]
```diff
@@ -21,11 +22,12 @@ class OwnerCannotShipError(Exception):
 class NoMergeCommentError(Exception):
     pass
 
+logger = logging.getLogger()
 
 def check_required_fields(required, **args):
     for i in required:
         if args[i] is None:
-            print 'Error: %s parameter not set' % i
+            logging.error(' %s parameter not set', i.upper())
```
Comment (cchristous) [inline github_pr.py]: all of your `logging.<level>()` calls need to be `logger.<level>()`. Otherwise, they use the default logging config

Comment (babineaum) [inline github_pr.py]: i changed this, but im now seeing that none of my logging is working
my guess is because logger isnt defined in the other methods.

Comment (dx-pbuckley) [inline github_pr.py]: I think you might be missing something like this: https://github.com/dataxu/jenkins-tools/blob/1369b1f0c3b7e9caf89dcfede446cc58668e61f8/scripts/blackout_dates/blackout_dates_calendar.py#L179

File: github_pr.py [hunk 4]
```diff
@@ -441,8 +460,22 @@ def main():
     parser.add_argument('--condition-approved-mergers', default=None, nargs='+', help='list of usernames of approved mergers')
     parser.add_argument('--condition-approved-mergers-file', action='store_true', help='check a file for list of usernames of approved mergers')
     parser.add_argument('--approved-mergers-file-path', default='./MAINTAINERS.txt', help='location of file of usernames of approved mergers')
+    parser.add_argument('-v', '--verbose', const=1, default=0, type=int, nargs="?",
+                    help="Logger verbosity: 0 = WARN (default), 1 = INFO, 2 = DEBUG")
 
     args = vars(parser.parse_args())
```
Comment (cchristous) [inline github_pr.py]: I'd recommend using 
```python
logging.basicConfig(level=level, format="%(asctime)s %(levelname)-7s %(filename)20s:%(lineno)-4d | %(message)s")```
You shouldn't need to do anything stuff with it

Comment (babineaum) [inline github_pr.py]: that is much nicer looking :) thanks

**Comment by babineaum** [comment]
@cchristous would love another look on the logging. thanks for your help with it, im taking some lessons right now that are starting to go through logging.

**Comment by cchristous** [comment]
looks fine

**Pull request merged**